File: client/src/components/financial/financialValueDefinitions.ts
export type ValueFormat = 'currency' | 'percent' | 'ratio'

export interface ValueDefinition {
  key: string
  label: string
  format: ValueFormat
}

export const BALANCO_VALUES: ValueDefinition[] = [
  { key: 'ATIVO_CIRCULANTE', label: 'Ativo Circulante', format: 'currency' },
  { key: 'ATIVO_NAO_CIRCULANTE', label: 'Ativo Não Circulante', format: 'currency' },
  { key: 'ATIVO_TOTAL', label: 'Ativo Total', format: 'currency' },
  { key: 'PASSIVO_CIRCULANTE', label: 'Passivo Circulante', format: 'currency' },
  { key: 'PASSIVO_NAO_CIRCULANTE', label: 'Passivo Não Circulante', format: 'currency' },
  { key: 'PASSIVO_TOTAL', label: 'Passivo Total', format: 'currency' },
  { key: 'PATRIMONIO_LIQUIDO', label: 'Patrimônio Líquido', format: 'currency' },
]

export const DRE_VALUES: ValueDefinition[] = [
  { key: 'RECEITA_TOTAL', label: 'Receita Total', format: 'currency' },
  { key: 'CUSTO_TOTAL', label: 'Custo Total', format: 'currency' },
  { key: 'DESPESA_TOTAL', label: 'Despesa Total', format: 'currency' },
  { key: 'DEPRECIACAO_AMORTIZACAO_TOTAL', label: 'Depreciação e Amortização', format: 'currency' },
]

export const INDICADORES: ValueDefinition[] = [
  { key: 'RESULTADO_PERIODO', label: 'Resultado do Período', format: 'currency' },
  { key: 'RESULTADO_ANTES_DEPRECIACAO_AMORTIZACAO', label: 'Resultado Antes de Depreciação e Amortização', format: 'currency' },
  { key: 'MARGEM_RESULTADO', label: 'Margem de Resultado', format: 'percent' },
  { key: 'LIQUIDEZ_CORRENTE', label: 'Liquidez Corrente', format: 'ratio' },
  { key: 'INDICE_ENDIVIDAMENTO', label: 'Índice de Endividamento', format: 'ratio' },
]

export function formatFinancialValue(value: number | undefined, format: ValueFormat): string {
  if (value === undefined) return '—'
  if (format === 'currency') return value.toLocaleString('pt-BR', { style: 'currency', currency: 'BRL' })
  if (format === 'percent') return `${value.toLocaleString('pt-BR', { maximumFractionDigits: 2 })}%`
  return value.toLocaleString('pt-BR', { maximumFractionDigits: 4 })
}

File: server/src/CreditScanAI.Api/Services/FinancialValueDefinitions.cs
using System.Globalization;

namespace CreditScanAI.Api.Services;

public enum FinancialValueFormat
{
    Currency,
    Percent,
    Ratio
}

public sealed record FinancialValueDefinition(string Key, string Label, FinancialValueFormat Format);

/// <summary>
/// Server-side mirror of client/src/components/financial/financialValueDefinitions.ts -
/// used by the PDF/Excel export (Fase 10) so the exported file uses the same
/// labels/grouping/formatting as the Dashboard and Consolidação pages. Keep
/// both in sync if either changes.
/// </summary>
public static class FinancialValueDefinitions
{
    public static readonly IReadOnlyList<FinancialValueDefinition> Balanco =
    [
        new(CalculationKeys.AtivoCirculante, "Ativo Circulante", FinancialValueFormat.Currency),
        new(CalculationKeys.AtivoNaoCirculante, "Ativo Não Circulante", FinancialValueFormat.Currency),
        new(CalculationKeys.AtivoTotal, "Ativo Total", FinancialValueFormat.Currency),
        new(CalculationKeys.PassivoCirculante, "Passivo Circulante", FinancialValueFormat.Currency),
        new(CalculationKeys.PassivoNaoCirculante, "Passivo Não Circulante", FinancialValueFormat.Currency),
        new(CalculationKeys.PassivoTotal, "Passivo Total", FinancialValueFormat.Currency),
        new(CalculationKeys.PatrimonioLiquido, "Patrimônio Líquido", FinancialValueFormat.Currency)
    ];

    public static readonly IReadOnlyList<FinancialValueDefinition> Dre =
    [
        new(CalculationKeys.ReceitaTotal, "Receita Total", FinancialValueFormat.Currency),
        new(CalculationKeys.CustoTotal, "Custo Total", FinancialValueFormat.Currency),
        new(CalculationKeys.DespesaTotal, "Despesa Total", FinancialValueFormat.Currency),
        new(CalculationKeys.DepreciacaoAmortizacaoTotal, "Depreciação e Amortização", FinancialValueFormat.Currency)
    ];

    public static readonly IReadOnlyList<FinancialValueDefinition> Indicadores =
    [
        new(CalculationKeys.ResultadoPeriodo, "Resultado do Período", FinancialValueFormat.Currency),
        new(CalculationKeys.ResultadoAntesDepreciacaoAmortizacao, "Resultado Antes de Depreciação e Amortização", FinancialValueFormat.Currency),
        new(CalculationKeys.MargemResultado, "Margem de Resultado", FinancialValueFormat.Percent),
        new(CalculationKeys.LiquidezCorrente, "Liquidez Corrente", FinancialValueFormat.Ratio),
        new(CalculationKeys.IndiceEndividamento, "Índice de Endividamento", FinancialValueFormat.Ratio)
    ];

    public static string Format(decimal? value, FinancialValueFormat format)
    {
        if (value is null)
        {
            return "—";
        }

        var ptBr = CultureInfo.GetCultureInfo("pt-BR");
        return format switch
        {
            FinancialValueFormat.Currency => value.Value.ToString("C", ptBr),
            FinancialValueFormat.Percent => $"{value.Value.ToString("N2", ptBr)}%",
            _ => value.Value.ToString("N4", ptBr)
        };
    }
}
